test/net/sourceforge/kolmafia/objectpool/ConcoctionTest.java

```diff
@@ -2,24 +2,46 @@
 
 import static org.junit.jupiter.api.Assertions.*;
 
+import java.util.HashMap;
+import java.util.Map;
 import net.java.dev.spellcast.utilities.LockableListModel;
 import net.sourceforge.kolmafia.persistence.ConcoctionDatabase;
-import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 
 /* This test was triggered by a runtime error traced back to sorting usable concoctions that
-said "Comparison method violates its general contract!"  It is likely that the error was caused by
-dynamic data, i.e. something that changed because a character left Ronin, but it can't hurt to verify
-the comparison against static data.
+said "Comparison method violates its general contract!"
  */
 
 public class ConcoctionTest {
 
+  static final Map<Integer, Concoction> idsAndMaps = new HashMap<>();
```

Comment:
Why not store this in an ArrayList or an array with true constant (and much lower overhead) random access?

Reply:
It seemed like a good idea at the time when there was also a corresponding Map<Concoction, Integer> so I could go back and forth between a Concoction and an integer.  I thought I was going to need Concoction as an index.  It doesn't look like I do.  Will revisit.